common/id/src/main/kotlin/com/cosmotech/api/id/hashids/HashidsCsmIdGenerator.kt

```diff
@@ -14,14 +17,33 @@ private const val ALPHABET = "abcdefghijklmnopqrstuvwxyz0123456789"
 @Component
 @ConditionalOnProperty(
     name = ["csm.platform.id-generator.type"], havingValue = "hashid", matchIfMissing = true)
-internal class HashidsCsmIdGenerator : AbstractCsmIdGenerator() {
+internal class HashidsCsmIdGenerator() : AbstractCsmIdGenerator() {
+
+  private val logger = LoggerFactory.getLogger(HashidsCsmIdGenerator::class.java)
+
+  private var encodedElementSupplier: (() -> ULong)? = null
+
+  constructor(encodedElementSupplier: () -> ULong) : this() {
```

Comment:
- when passed to this secondary constructor, I expect it to be non-null. It is null otherwise if calling the default constructor.
- the supplier is passed at this time in the tests, to have control over the number to encode (and test values higher than the max allowed): [HashidsCsmIdGeneratorTests](https://github.com/Cosmo-Tech/cosmotech-api/blob/32ac5da378/common/id/src/test/kotlin/com/cosmotech/api/id/hashids/HashidsCsmIdGeneratorTests.kt#L22)